src/main/java/frc/robot/subsystems/elevator/Elevator.java

```diff
@@ -1,8 +1,108 @@
 package frc.robot.subsystems.elevator;
+
+import java.util.Optional;
+import java.util.TreeMap;
+
+import com.ctre.phoenix6.configs.Slot0Configs;
+import com.ctre.phoenix6.configs.TalonFXConfiguration;
+import com.ctre.phoenix6.configs.TalonFXConfigurator;
+import com.ctre.phoenix6.controls.Follower;
+import com.ctre.phoenix6.controls.PositionVoltage;
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.ctre.phoenix6.signals.InvertedValue;
+import com.ctre.phoenix6.signals.NeutralModeValue;
+
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.controller.SimpleMotorFeedforward;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.util.sendable.SendableBuilder;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.drive.RobotDriveBase.MotorType;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
-import frc.robot.Constants;
 import frc.robot.RobotMap;
 
 public class Elevator extends SubsystemBase {
-    
+    public enum ElevatorHeight {
+        HOME,
+        SCORE_L4,
+        SCORE_L3,
+        SCORE_L2,
+        SCORE_L1,
+        PRE_SHALLOWCLIMB,
+        SCORE_SHALLOWCLIMB;
+    }
+
+   
+    private TreeMap<ElevatorHeight, Double> elevatorPositions = new TreeMap<ElevatorHeight, Double>();
+                
+    private ElevatorHeight desiredPosition = ElevatorHeight.HOME;
+
+    public TalonFX leftMotor = new TalonFX(RobotMap.LEFT_ELEVATOR_CAN_ID);
+    public TalonFX rightMotor = new TalonFX(RobotMap.RIGHT_ELEVATOR_CAN_ID);
```

Comment:
these should be private. that way, no methods outside this class can command input to the motors (without needing to use setDesiredPosition).